utils/domainstate.ml.c

```diff
@@ -0,0 +1,37 @@
+/**************************************************************************/
+/*                                                                        */
+/*                                 OCaml                                  */
+/*                                                                        */
+/*      KC Sivaramakrishnan, Indian Institute of Technology, Madras       */
+/*                 Stephen Dolan, University of Cambridge                 */
+/*                                                                        */
+/*   Copyright 2019 Indian Institute of Technology, Madras                */
+/*   Copyright 2019 University of Cambridge                               */
+/*                                                                        */
+/*   All rights reserved.  This file is distributed under the terms of    */
+/*   the GNU Lesser General Public License version 2.1, with the          */
+/*   special exception on linking described in the file LICENSE.          */
+/*                                                                        */
+/**************************************************************************/
+
+#define CAML_CONFIG_H_NO_TYPEDEFS
+#include "config.h"
```

Comment:
This looks quite fragile, and requires extra care in config.h to avoid generating in text.  Why do we need config.h at all?

Couldn't we generate the .ml/.mli files with grep/awk instead?

Reply:
You are right. config.h is not needed. I'll remove it and push a commit.

For the code generation, I am OK with either. Perhaps @stedolan has an opinion since he wrote this code.